printing/printer.ml

```diff
@@ -751,63 +754,38 @@ let default_pr_subgoals ?(pr_first=true)
     | None -> ()
   in
+
+  (** Main function *)
   match goals with
   | [] ->
       begin
-	let exl = Evd.undefined_map sigma in
-	if Evar.Map.is_empty exl then
-	  (str"No more subgoals." ++ print_dependent_evars None sigma seeds)
-	else
+        let exl = Evd.undefined_map sigma in
+        if Evar.Map.is_empty exl then
+          (str"No more subgoals." ++ print_dependent_evars None sigma seeds)
+        else
           let pei = pr_evars_int sigma ~shelf ~givenup:[] 1 exl in
-	  v 0 ((str "No more subgoals,"
-                ++ str " but there are non-instantiated existential variables:"
-	        ++ cut () ++ (hov 0 pei)
-                ++ print_dependent_evars None sigma seeds
-                ++ cut () ++ str "You can use Grab Existential Variables."))
+          v 0 ((str "No more subgoals,"
+                      ++ str " but there are non-instantiated existential variables:"
+                      ++ cut () ++ (hov 0 pei)
+                      ++ print_dependent_evars None sigma seeds
+                      ++ cut () ++ str "You can use Grab Existential Variables."))
       end
   | g1::rest ->
       let goals = print_multiple_goals g1 rest in
       let ngoals = List.length rest+1 in
       v 0 (
-	int ngoals ++ focused_if_needed ++ str(String.plural ngoals "subgoal")
+	      int ngoals ++ focused_if_needed ++ str(String.plural ngoals "subgoal")
         ++ print_extra
         ++ str (if (should_gname()) then ", subgoal 1" else "")
         ++ (if should_tag() then pr_goal_tag g1 else str"")
         ++ pr_goal_name sigma g1 ++ cut () ++ goals
         ++ (if unfocused=[] then str ""
            else (cut() ++ cut() ++ str "*** Unfocused goals:" ++ cut()
                  ++ pr_rec (List.length rest + 2) unfocused))
-	++ print_dependent_evars (Some g1) sigma seeds
+                 ++ print_dependent_evars (Some g1) sigma seeds
       )
 
-(**********************************************************************)
-(* Abstraction layer                                                  *)
-
-
-type printer_pr = {
- pr_subgoals            : ?pr_first:bool -> Pp.t option -> evar_map -> seeds:goal list -> shelf:goal list -> stack:int list -> unfocused:goal list -> goals:goal list -> Pp.t;
- pr_subgoal             : int -> evar_map -> goal list -> Pp.t;
- pr_goal                : goal sigma -> Pp.t;
-}
-
-let default_printer_pr = {
- pr_subgoals = default_pr_subgoals;
- pr_subgoal  = default_pr_subgoal;
- pr_goal     = default_pr_goal;
-}
-
-let printer_pr = ref default_printer_pr
-
-let set_printer_pr = (:=) printer_pr
-
-let pr_subgoals ?pr_first x = !printer_pr.pr_subgoals ?pr_first x
-let pr_subgoal  x = !printer_pr.pr_subgoal  x
-let pr_goal     x = !printer_pr.pr_goal     x
-
-(* End abstraction layer                                              *)
-(**********************************************************************)
-
-let pr_open_subgoals ~proof =
+let pr_open_subgoals ~proof ?(quiet=false) ?(diffs=None) () =
```

Comment:
revert the order so you don't need to pass () in all the functions below.

Reply:
Revert?? Do you mean just put the quiet and diffs parameters before proof?

Reply:
Revert means "return to (a previous state, practice, topic, etc.)."
"he reverted to his native language"
synonyms: | return, go back, change back, default

Reply:
I spent an hour or two trying to get this to work.  I think it is not supported.  The OCaml manual says 

```
A function taking some optional arguments must also take at least one non-optional argument.
The criterion for deciding whether an optional argument has been omitted is the non-labeled
application of an argument appearing after this optional argument in the function type
```

Either I put in `()` or make proof unlabeled.  FWIW, it seemed the compiler was getting confused at vernacentries.ml:1931--it would report a message seemingly related to pr_open_subgoals, but the location information pointed to the next line.